tern/analyze/docker/dockerfile.py

```diff
@@ -118,6 +126,38 @@ def parse_from_image(dfobj):
     return image_list
 
 
+def expand_from_images(dfobj):
+    '''Replace all parent_image values with their full tag@sha:digest value.
+    Update the structure dictionary with the same information.
+    dfobj: the Dockerfile object created using get_dockerfile_obj'''
+    # update parent_images
+    images = parse_from_image(dfobj)
+    for i, img in enumerate(images):
+        dfobj.parent_images[i] = find_image_digest(img['name'], img['tag'])
+    # update structure
+    for i, command_dict in enumerate(dfobj.structure):
+        if command_dict['instruction'] == 'FROM':
+            dfobj.structure[i]['content'] = 'FROM ' + find_image_digest(
+                command_dict['value'].split(tag_separator, 2)[0],
+                command_dict['value'].split(tag_separator, 2)[1])
+            dfobj.structure[i]['value'] = find_image_digest(
+                command_dict['value'].split(tag_separator, 2)[0],
+                command_dict['value'].split(tag_separator, 2)[1])
```

Comment:
I would rather pass in the results from `find_image_digest`. Either that or move the whole function into `container.py`. I'm mostly thinking in terms of the dependency on the external image for testing purposes. If this gets moved to `container.py` then we can use the test image(s) on Dockerhub for all of those tests.

Reply:
Pass the results of `find_image_digest` to the `expand_from_images` instead of the dfobj or in addition?

Reply:
I also think that this should stay in `dockerfile.py`, as it is a parallel function to `expand_vars` and will only be used in the dockerfile freeze functionality.

Reply:
OK. If we need to move it somewhere else, we can do that.